drivers/platform/aducm3029/uart.c

```diff
@@ -288,14 +289,19 @@ int32_t uart_read(struct uart_desc *desc, uint8_t *data,
 	while (extra->read_desc.is_nonblocking)
 		;
 
-	if (ADI_UART_SUCCESS != adi_uart_Read(
-		    (ADI_UART_HANDLE const)extra->uart_handler,
-		    (void *const)data,
-		    (uint32_t const)bytes_number,
-		    bytes_number > 4 ? true : false,
-		    &errors))
-		goto failure;
-
+	idx = 0;
+	while (bytes_number) {
```

Comment:
It seems to me, that this function is becoming blocking now, regardless of `extra->read_desc.is_nonblocking` value

Reply:
This function is always blocking. There is uart_read/write_nonblocking for nonblocking operations.
The extra->read_desc.is_nonblocking is a flag used to know if a nonblocking is executed. And wait for it to start a new transaction.  A blocking transaction can't begin if a nonblocking is in process.